storage/src/vespa/storage/distributor/pending_bucket_space_db_transition.h

```diff
@@ -0,0 +1,92 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+#pragma once
+
+#include "pending_bucket_space_db_transition_entry.h"
+#include <vespa/storage/bucketdb/bucketdatabase.h>
+#include <unordered_set>
+
+namespace storage::api { class RequestBucketInfoReply; }
+namespace storage::lib { class ClusterState; }
+
+namespace storage::distributor {
+
+class ClusterInformation;
+class PendingClusterState;
+
+/**
+ * Class used by PendingClusterState to track request bucket info
+ * reply result within a bucket space and apply it to the distributor
+ * bucket database when switching to the pending cluster state.
+ */
+class PendingBucketSpaceDbTransition : public BucketDatabase::MutableEntryProcessor
+{
+public:
+    using Entry = pendingbucketspacedbtransition::Entry;
```

Comment:
This is a bit of a mouthful 😉  Maybe we could shorten it down to e.g. `dbtransition::Entry`? Type-wise, it's really just a tuple of bucket id and bucket info for that bucket, which could theoretically be used elsewhere as well